src/generators/validators.cr

```diff
@@ -0,0 +1,23 @@
+module LuckyCli::Generators::Validators
+  macro validate(name, &block)
+    module {{ name }}
+      extend self
+
+      def sanitize(name)
+        {{ yield(name) }}
+      end
+
+      def valid?(name)
+        sanitize(name) == name
+      end
+    end
+  end
+
+  validate ProjectName do |name|
+    name.downcase.gsub(/[^a-z0-9_-]/,"_").strip('_')
+  end
+
+  validate FolderName do |name|
+    name.gsub(/[^a-zA-Z0-9_-]/,"_").strip('_')
+  end
+end
```

Comment:
If it is just a project name validator then this can probably be done without macros, and instead some class methods on a class or maybe a module with `extend self`?